pcf/particle/aws/cloudfront/cloudfront.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pcf.core.aws_resource import AWSResource
+from pcf.core import State
+from pcf.util import pcf_util
+
+
+class CloudFront(AWSResource):
+    """
+    Particle that maps to AWS Cloud Front
+    """
+
+    flavor = 'cloudfront'
+
+    START_PARAM_FILER = {
+        'CallerReference',
+        'Aliases',
+        'DefaultRootObject',
+        'Origins',
+        'OriginGroups',
+        'DefaultCacheBehavior',
+        'CacheBehaviors',
+        'CustomErrorResponses',
+        'Comment',
+        'Logging',
+        'PriceClass',
+        'Enabled',
+        'ViewerCertificate',
+        'Restrictions',
+        'WebACLId',
+        'HttpVersion',
+        'IsIPV6Enabled'
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0,
+    }
+
+    UNIQUE_KEYS = ["aws_resource.Comments"]
+
+    def __init__(self, particle_definition):
+        """
+        Args:
+            particle_definition (definition): desired configuration of the particle
+        """
+        super(CloudFront, self).__init__(particle_definition, 'cloudfront')
+        self._id = ""
+
+    def _set_unique_keys(self):
+        """
+        Logic that sets keys from state definition that are used to uniquely identify the distribution
+        """
+        self.unique_keys = CloudFront.UNIQUE_KEYS
+
+    def _start(self):
+        """
+        Creates the distribution according to the particle definition and adds tags if necessary
+        Returns:
+            hosted_zone: boto3 response
+        """
+        start_definition = pcf_util.param_filter(self.desired_state_definition, CloudFront.START_PARAM_FILER)
+        response = self.client.create_distribution_with_tags(
+            DistributionConfigWithTags={
+                "DistributionConfig": start_definition,
+                "Tags": {
+                    "Items": self.desired_state_definition.get("Tags")
```

Comment:
does the distribution get created as expected if `"Items": None` when tags aren't specified? At the very least I would think this should default to an empty list instead of `None`